pkg/lib/cockpit/location.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { url_root, calculate_application } from './_internal/location-utils';
+
+/* HACK: Mozilla will unescape 'window.location.hash' before returning
+* it, which is broken.
+*
+* https://bugzilla.mozilla.org/show_bug.cgi?id=135309
```

Comment:
What's the actual hack here? Also, that bug was fixed 21 (!) years ago.

Reply:
Right, I actually tested this and thought about a `test-server` test for it.

```
window.location.href.split('#')[1] == window.location.hash.slice(1) 
```

I am happy to do this as a follow up but I don't want to go crazy with possibly interested regressions / failures :)

Reply:
The hack is to not use location.hash, but to split it out of location.href.

Reply:
Seems to be fixed indeed, but it sure wasn't at the time we introduced this hack, which was less than 21 years ago. :-)

Reply:
Fixed in https://github.com/cockpit-project/cockpit/pull/21382

Reply:
Will rebase this PR afterwards.